### ISSUE TITLE
Black spaces

### ISSUE THREAD

**Issue by Samamax** [AmbitiousPilots/LittleNavmapOFMTheme#18]
Hi,
even though I have updated AIRAC, *openflightmap the map image disappears and it remain black spaces. Why? Thanks and good day.

![ImageOpenFlightMap](https://user-images.githubusercontent.com/103597897/163238754-129d5338-4368-4317-adf7-b37b7d2e2050.png)
T

**Comment by AmbitiousPilots** [comment]
The map cache needs to be cleared in order for the new ones to be loaded. You can do this in LittleNavmap directly or remove the cached files manually.

**In LittleNavmap:**
Go to: Tools -> Options -> Cache and Files, then click on "Clear Disk Cache".

**Manually:**
Remove all contents in the following dir and restart LittleNavmap:
`%localappdata%\.marble\data\maps\earth\openflightmaps`

**Comment by rippy65** [comment]
Hi,

similar problem here: AIRAC Cycle 2203 rev.1, latest LNM, latest LNMOFMTheme, all installed according to instructions. Only black maps where they shouldn't (e.g. around EDDS, as shown in the screenshot)
![image](https://user-images.githubusercontent.com/58223079/163379762-9949b262-8d99-4d73-8dd4-177bc0697531.png)

**Comment by jonaseberle** [comment]
@rippy65 please report if you have you done what @AmbitiousPilots suggested.

**Comment by rippy65** [comment]
> @rippy65 please report if you have you done what @AmbitiousPilots suggested.

sure. Cache cleared before that screenshot.

**Comment by AmbitiousPilots** [comment]
Works fine here (LittleNavmap v2.6.18, LittleNavmapOFMTheme v1.0.14):
![image](https://user-images.githubusercontent.com/72250898/163383524-f4395870-dc0b-4ee6-a95e-1bebcc641652.png)

**Comment by rippy65** [comment]
> Works fine here (LittleNavmap v2.6.18, LittleNavmapOFMTheme v1.0.14): ![image](https://user-images.githubusercontent.com/72250898/163383524-f4395870-dc0b-4ee6-a95e-1bebcc641652.png)

Glaubischo :) ...I've setup OFMTheme on 3 different boxes - always and everywhere black...I must be doing wrong - but what? Are there any win11 firewall exceptions necessary?

**Comment by AmbitiousPilots** [comment]
Yes, when the cache is cleared and it's still not loading the map tiles it might indeed be a firewall blocking the traffic. Make sure to allow network access for LittleNavmap. The map tiles are loaded from the following domain, so try to white list it when possible: 

`https://nwy-tiles-api.prod.newaydata.com`

**Comment by rippy65** [comment]
Thx - I'll try that shortly. Meanwhile I tried an LNM/OFMtheme install on another laptop (I have a lot here... ;) ) - and there OFM showed up "out of the box")...will have to check the different install processes for 100% consistency.

I'll report here ASAP.

**Comment by Samamax** [comment]
> The map cache needs to be cleared in order for the new map tiles to be loaded. You can do this in LittleNavmap directly or remove the cached files manually.
> 
> **In LittleNavmap:** Go to: Tools -> Options -> Cache and Files, then click on "Clear Disk Cache".
> 
> **Manually:** Remove all contents in the following dir and restart LittleNavmap: `%localappdata%\.marble\data\maps\earth\openflightmaps`

I cleared the cache from the disk and disabled antivirus and firewall. I'm sorry the problem remains

**Comment by albar965** [comment]
I'd recommend to clear the cache manually from the disk. The clear cache command in LNM uses the Marble widget function which takes some time running in the background.

Clearing manually (close LNM before) also allows you to keep other cached map files like OpenStreetMap.

**Comment by Samamax** [comment]
> I'd recommend to clear the cache manually from the disk. The clear cache command in LNM uses the Marble widget function which takes some time running in the background.
> 
> Clearing manually (close LNM before) also allows you to keep other cached map files like OpenStreetMap.

I tried, I clerared manually and it didn't work.
![imagine2](https://user-images.githubusercontent.com/103597897/163408862-8dda414c-5dd2-4704-97e6-2ce742cb3e96.png)

**Comment by rippy65** [comment]
> Thx - I'll try that shortly. Meanwhile I tried an LNM/OFMtheme install on another laptop (I have a lot here... ;) ) - and there OFM showed up "out of the box")...will have to check the different install processes for 100% consistency.
> 
> I'll report here ASAP.

Ok...I have OFMTheme now up and running (=showing) on the FS Computer aswell as on different networked machines.

On all FAILED setups there was a coincidence:  I had LNM installed in my /username/documents folder, which was synced by MS OneDrive over the internet with all my boxes inside my private network. 

So I had an idea: Maybe the fact that OneDrive syncs my /documents folder gave issues... anyway : I deleted all LNM folders (also Username/AppData/Local/.marble AND Username/Appdata/Roaming/ABarthel) and did a fresh LNM install inside a new dir D:/FSData (which is not synced by OneDrive). Voila... - THAT worked. OFMTheme (aswell as all others...) is showing up perfectly.

Phew. ;)

Thanks to all who tried to help!

**Comment by Samamax** [comment]
> > I'd recommend to clear the cache manually from the disk. The clear cache command in LNM uses the Marble widget function which takes some time running in the background.
> > Clearing manually (close LNM before) also allows you to keep other cached map files like OpenStreetMap.
> 
> I tried, I cleared manually and it didn't work. ![imagine2](https://user-images.githubusercontent.com/103597897/163408862-8dda414c-5dd2-4704-97e6-2ce742cb3e96.png)

Isn't there another solution? The black spaces remained!

**Comment by Samamax** [comment]
> > > I'd recommend to clear the cache manually from the disk. The clear cache command in LNM uses the Marble widget function which takes some time running in the background.
> > > Clearing manually (close LNM before) also allows you to keep other cached map files like OpenStreetMap.
> > 
> > 
> > I tried, I cleared manually and it didn't work. ![imagine2](https://user-images.githubusercontent.com/103597897/163408862-8dda414c-5dd2-4704-97e6-2ce742cb3e96.png)
> 
> Isn't there another solution? The black spaces remained!


To clear the cache in .maple isn't the solution.  How to do?

**Comment by janus45** [comment]
I have the same issue the map keeps dark.

**Comment by f7540w** [comment]
Problem back here also - on two different networked boxes...As I didn't change anything on them since I solved the "black maps" issue on April 14th, I suspect it has something to do with a code change from MS/Asobo (latest mandatory update - > "TG Maverick"). This update fixed another bug on my MSFS setups - exe.xml not loading - but presumably broke the through littlenavconnect perfectly working OFM . Or anything has changed on the OFM server side...? Any clues, anyone?

**Comment by AndreGeddert** [comment]
Ok, i took a look at this again. Seems the problem was only with the base-tiles (OFM is dual-layered, "base" and "aero" map). So i deleted the cached base-tiles, in my case located in C:\Users\andre\AppData\Local\.marble\data\maps\earth\openflightmaps 
i kept the aero Folder in there because aero-tiles are not affected by this issue.
When i start LNM again all tiles are loaded correctly. Maybe rendering of the 2205 base tiles look longer than usual this time. But now it look like everything is available. 
So clear you caches manually (while LNM is not running) and try again.
(@rippy65 , MS/Asobo Updates do now affect this map as this i completely independent)

**Comment by f7540w** [comment]
Mhmmmm...I know it sounds like a dumb excuse, but: I didn't change anything on the networked boxes where lnm with ofm is running - but OFM just doesn't work anymore. Well, ATM I'm flying without them. No big problem.

**Comment by AndreGeddert** [comment]
Ok, sometimes i noticed that something could be wrong in the LNM config/profile that leads to strange behaviour. So i suggest to make a backup and then delete the folder C:\Users\andre\AppData\Roaming\ABarthel (yours will be "%appdata%\ABarthel"), and then try again with a fresh config.

**Comment by f7540w** [comment]
@AndreGeddert: Didn't help. My boxes are backupped on a daily basis - and 8 days ago (the last time I fired up MSFS) LNM with OFM was working fine. So I restored that Macrium-BU, but...black maps.

It must be something other. 

But ATM I won't track the issue (again...) further down, because I don't have the time due to RL workload - and LNM is principally working fine both on the MSFS rig aswell as on networked machines. 

It's just OFM which won't work - and given the circumstances I can live with that. In 2 years I'm flightsimming for 40 years: In this time I learned that something is always broken, and most likely that "something" will work again...sometimes, somewhere 😎😉. 

Anyway, thanks for your kind try to help!

**Comment by f7540w** [comment]
> Anyway, thanks for your kind try to help!

Problem solution: I edited openflightmaps.dgml to reflect the current airac cycle. Now all is fine again. See https://forums.flightsimulator.com/t/vfr-map-for-little-navmap/317879/91?u=mager1965

**Comment by snipem** [comment]
This is fixed in #19 and the upstream. Maybe you should include a link to https://github.com/AmbitiousPilots/LittleNavmapOFMTheme/archive/refs/heads/main.zip instead of individual releases. I ran into the same issue and downloaded an old release.

**Comment by Samamax** [comment]
Hi, with the last version 2.6.19 reappeared the problems with OpenFlightMap
also updating AIRAC in all documents. The screen  always black.


Il giorno dom 26 giu 2022 alle ore 11:38 Matthias Küch <
***@***.***> ha scritto:

**Comment by DigiOrti** [comment]
Same here, but with editing the AIRAC cycle, I coud it get back to work.

Two times change the query to:
query=“path=2208/base/latest”

**Comment by MedicoreV8** [comment]
Seems like there is a problem with OpenFlightMaps with all AIRAC cycles right now.
I found out when I tried updating to cycle 2209 and found out, that even after clearing the cache in LNMP the tiles remained black. The live map on their website shows black too. Is anyone else experiencing this too?
(https://www.openflightmaps.org/eh-netherlands/)

**Comment by DigiOrti** [comment]
> Seems like there is a problem with OpenFlightMaps with all AIRAC cycles right now. I found out when I tried updating to cycle 2209 and found out, that even after clearing the cache in LNMP the tiles remained black. The live map on their website shows black too. Is anyone else experiencing this too? (https://www.openflightmaps.org/eh-netherlands/)

Well, if you AIRAC matches, you don't have problems.

**Comment by MedicoreV8** [comment]
![E5DAFAD1-1272-4DB5-BBD2-B62A480BC661](https://user-images.githubusercontent.com/32266563/189207816-82fa5bb4-2928-4e29-a64a-ddc270464273.png)
[>](![5A9B7599-2402-4588-A1D3-F09B9BEA3397](https://user-images.githubusercontent.com/32266563/189207665-393df81b-5db7-46e0-89ce-66a0a7fb3b9c.png)) > Seems like there is a problem with OpenFlightMaps with all AIRAC cycles right now. I found out when I tried updating to cycle 2209 and found out, that even after clearing the cache in LNMP the tiles remained black. The live map on their website shows black too. Is anyone else experiencing this too? (https://www.openflightmaps.org/eh-netherlands/)
> 
> Well, if you AIRAC matches, you don't have problems.

Nope. As I have mentioned, even the website itself doesn’t show the map tiles for me anymore, regardless of the selected AIRAC in the top right. I don’t think this is user error with littlenavmap, the file or the cache.

see attached screenshot of the website (pc/cellphone wifi/4G doesn’t make a difference)

**Comment by jonaseberle** [comment]
I am using https://github.com/AmbitiousPilots/LittleNavmapOFMTheme/pull/20 (`.../latest/`) and that works fine for me. 
You might have to clear caches (Options -> Cache) to be able to test properly.

**Comment by AndreGeddert** [comment]
@AmbitiousPilots does anything speaks again merging #20 ?

**Comment by Samamax** [comment]
also the update of the airac 2209 always black screens!!!

Il giorno gio 15 set 2022 alle ore 15:51 Andre Geddert <
***@***.***> ha scritto:

**Comment by doitagainstudio** [comment]
Hello all. I'm having the issue as well. I tried passing the whole URL path on a browser and it gives black tiles (only the JPG for the base), so I guess the issue is server-sided and really needs to be troubleshot. Hope someone would find a fix ASAP :-(

**Comment by jonaseberle** [comment]
@doitagainstudio Have you applied https://github.com/AmbitiousPilots/LittleNavmapOFMTheme/pull/20/files? That works well for me since months now.

**Comment by MedicoreV8** [comment]
> Hello all. I'm having the issue as well. I tried passing the whole URL path on a browser and it gives black tiles (only the JPG for the base), so I guess the issue is server-sided and really needs to be troubleshot. Hope someone would find a fix ASAP :-(

Yeah, It has been mentioned. Applying the changes in merge #20 did the trick for me.

**Comment by doitagainstudio** [comment]
Working now, thx! 😀

Il lun 19 set 2022, 11:21 Jonas Eberle ***@***.***> ha
scritto:

**Comment by Samamax** [comment]
Hi,
the only thing to do is to update AIRAC. If you have Navigraph it's
possible  update Airac automatically  LittleNavMap. However, even with this
solution, the black area remain the same if you increase the size of the
map.
Good luck

Il giorno sab 28 mag 2022 alle ore 11:59 janus45 ***@***.***>
ha scritto:

**Comment by jonaseberle** [comment]
@Samamax this is not related to the nav data from Navigraph. 

If you have applied the changes from https://github.com/AmbitiousPilots/LittleNavmapOFMTheme/issues/18#issuecomment-1250776475 and have still issues, please attach a screenshot.

Here is an example of how it should look like:
![image](https://user-images.githubusercontent.com/1678001/195034898-b640a49b-a59b-4b7c-8940-648d8ec724e8.png)

The black area is Great Britain where no openflightmaps data is available.
